RecoEcal/EgammaClusterProducers/src/PFECALSuperClusterProducer.cc

```diff
@@ -159,6 +159,10 @@ PFECALSuperClusterProducer::PFECALSuperClusterProducer(const edm::ParameterSet&
 
 PFECALSuperClusterProducer::~PFECALSuperClusterProducer() {}
 
+void PFECALSuperClusterProducer::beginRun(Run const& run, EventSetup const& iE) {
+  superClusterAlgo_.updateSCParams(iE);
```

Comment:
If the EventSetup objects are used only in the `produce()` function, the recommended pattern is to get the products only in the `produce()` function. Then the `ESGetToken`s won't need `edm::Transition::BeginRun` either.